wagtail/contrib/frontend_cache/utils.py

```diff
@@ -146,6 +146,112 @@ def purge_pages_from_cache(pages, backend_settings=None, backends=None):
         purge_urls_from_cache(urls, backend_settings, backends)
 
 
+def purge_site(
+    site,
+    backend_settings=None,
+    backends=None,
+    url_batch_chunk_size=2000,
+):
+    # Classify available backends, prioritising those that are capable
+    # of 'hostname-specific' purges, followed by those capabale of
+    # 'everything' purges
+    purge_hostname_backends = {}
+    purge_everything_backends = {}
+    other_backends = {}
+    for name, backend in get_backends(backend_settings, backends).items():
+        if backend.invalidates_hostname(site.hostname):
+            if backend.hostname_purge_supported():
+                purge_hostname_backends[name] = backend
+            elif backend.allow_everything_purge_for_hostname(site.hostname):
+                purge_everything_backends[name] = backend
+            else:
+                other_backends[name] = backend
+
+    if (
+        not purge_hostname_backends
+        and not purge_everything_backends
+        and not other_backends
+    ):
+        logger.info("Unable to find purge backend for %s", site.hostname)
+        return
+
+    for name, backend in purge_hostname_backends.items():
+        logger.info(f"Purging hostname via backend: {name}")
+        backend.purge_hostname(site.hostname)
+
+        # NOTE: We could potentially return `None` after the first purge here,
+        # but there's technically nothing to stop developers registering
+        # multiple backends for the same hostname, and no way for us to
+        # know what is redundant, or which backend is more important
+
+    for name, backend in purge_everything_backends.items():
+        # This will likely purge more than necessary, but will be more efficient
+        # than cycling through every page in the tree to generate a list of URLs
+
+        # NOTE: It is up to developers to disable 'everything' purges for
+        # backends where it is too disruptive to other services
+        logger.info(f"Purging everything via backend: {name}")
+        backend.purge_everything()
+
+        # NOTE: We could potentially return `None` after the first purge here,
+        # but there's technically nothing to stop developers registering
+        # multiple backends for the same hostname, and no way for us to
+        # know what is redundant, or which backend is more important
+
+    if other_backends:
+        # Count descendants
+        descendant_count = site.root_page.get_descendants().count()
+        logger.info(
+            "Purging %d page URLs for %s",
+            descendant_count,
+            site.hostname,
+        )
+        batch = PurgeBatch()
+        purge_fixed_paths_separately = False
```

Comment:
Question: What is "fixed"?

Reply:
Fixed paths are the paths that can be specified via the `WAGTAILFRONTENDCACHE_FIXED_SITE_PATHS` setting